qiskit/algorithms/amplitude_amplifiers/grover.py

```diff
@@ -156,31 +162,83 @@ def __init__(
         else:
             self._iterations = iterations
 
+        if quantum_instance is not None and sampler is not None:
+            raise ValueError("Only one of quantum_instance or sampler can be passed, not both!")
+
+        # check positionally passing the sampler in the place of quantum_instance
+        # which will be removed in future
+        if isinstance(quantum_instance, BaseSampler):
+            sampler = quantum_instance
+            quantum_instance = None
+
         self._quantum_instance = None
         if quantum_instance is not None:
-            self.quantum_instance = quantum_instance
+            warnings.warn(
+                "The quantum_instance argument has been superseded by the sampler argument. "
+                "This argument will be deprecated in a future release and subsequently "
+                "removed after that.",
+                category=PendingDeprecationWarning,
+                stacklevel=2,
+            )
+            with warnings.catch_warnings():
+                warnings.simplefilter("ignore", category=PendingDeprecationWarning)
+                self.quantum_instance = quantum_instance
+
+        self._sampler = sampler
 
         self._sample_from_iterations = sample_from_iterations
         self._iterations_arg = iterations
 
     @property
+    @deprecate_function(
+        "The Grover.quantum_instance getter is pending deprecation. "
+        "This property will be deprecated in a future release and subsequently "
+        "removed after that.",
+        category=PendingDeprecationWarning,
+    )
     def quantum_instance(self) -> Optional[QuantumInstance]:
-        """Get the quantum instance.
+        r"""Pending deprecation\: Get the quantum instance.
```

Comment:
```suggestion
        r"""Pending deprecation\; Get the quantum instance.
```
One last try - the : is still confusing the output so go with ; (semi-colon)!

![image](https://user-images.githubusercontent.com/40241007/191851907-5d50eab6-635f-45d1-8188-30f7e2ecd8e2.png)